lib/ansible/utils/plugin_docs.py

```diff
@@ -204,6 +205,16 @@ def add_fragments(doc, filename, fragment_loader, is_module=False):
         raise AnsibleError('unknown doc_fragment(s) in file {0}: {1}'.format(filename, to_native(', '.join(unknown_fragments))))
 
 
+def _handle_deprecated_kwarg(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        if len(args) >= 6 or 'is_module' in kwargs:
+            display.deprecated("is_module is deprecated, pass plugin_type='module' instead", version='2.19')
+        return func(*args, **kwargs)
+    return wrapper
+
+
+@_handle_deprecated_kwarg
 def get_docstring(filename, fragment_loader, verbose=False, ignore_errors=False, collection_name=None, is_module=None, plugin_type=None):
```

Comment:
You could simplify this by making the default value for `is_module` a sentinel (thanks @sivel for the suggestion).
```python
from ansible.utils.sentinel import Sentinel


def get_docstring(filename, fragment_loader, verbose=False, ignore_errors=False, collection_name=None, is_module=Sentinel, plugin_type=None):
    """
    DOCUMENTATION can be extended using documentation fragments loaded by the PluginLoader from the doc_fragments plugins.
    """
    if is_module is Sentinel:
        is_module = None
    else:
        display.deprecated("is_module is deprecated, pass plugin_type='module' instead", version='2.19')
```

Reply:
Done.

Reply:
Resolving